moto/datasync/urls.py

```diff
@@ -2,6 +2,6 @@
 
 from .responses import DataSyncResponse
 
-url_bases = ["https?://(.*?)(datasync)(.*?).amazonaws.com"]
+url_bases = [r"https?://(.*?)(datasync)\.(.*?).amazonaws.com"]
```

Comment:
It should be :
`url_bases = [r"https?://(.*\.)?(datasync)\.(.+).amazonaws.com"]` no?
I assume we want to include subdomains of datasync and not something ending with datasync?
And about the region, we should use (.+) as other services to ensure there is something between both dots.
 I'm not really familiar with this AWS service.

Reply:
Not too familiar with it either, so I'm not going to make too many assumptions about the URL formats :)